src/main/scala/dev/kamu/cli/MetadataRepository.scala

```diff
@@ -62,13 +63,36 @@ class MetadataRepository(
       yaml.load[Manifest[Dataset]](inputStream).content
     } catch {
       case e: Exception =>
-        logger.error(s"Error while loading data source manifest: $p")
+        logger.error(s"Error while loading data source manifest from file: $p")
         throw e
     } finally {
       inputStream.close()
     }
   }
 
+  def loadDatasetFromURI(uriStr: String): Dataset = {
+    val uri = new URI(uriStr)
+    uri.getScheme match {
+      case "https" => loadDatasetFromURL(uri.toURL)
+      case "http"  => loadDatasetFromURL(uri.toURL)
+      case "ftp"   => loadDatasetFromURL(uri.toURL)
```

Comment:
Reading `Source.fromURL` docs I doubt FTP source will work. In ingest-polling app I'm using a 3rd party FTP client, but here if it's not supported I wouldn't bother and just returned an error.